prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5291,8 +5291,9 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 // encode a closure deep copy or at least a stub
                 debug!("warning: ty::TyKind::Closure not implemented yet");
                 Vec::new()
-            }
-            ty::TyKind::Float(_) => unimplemented!("Unsure what to do here"),
+            }, 
+
+            ty::TyKind::Float(_) => unimplemented!("unsure what comes here"),
```

Comment:
This commit fixes the fold-unfold error: https://github.com/viperproject/prusti-dev/commit/9399c2b1432b9aaad3d2191eb287a930720e8ab9

I found what needs to be fixed by searching for `"unsupported"`.

Now the error is:
```
[2021-05-28T07:25:42Z ERROR viper::verifier] The verifier returned an unhandled error of type viper.silicon.reporting.Z3InteractionFailed$$anon$2: Interaction with Z3 (instance 08) failed: Unexpected output of Z3. Expected 'success' but found: (error "line 182 column 74: unknown function/constant $SortWrappers.___FloatingPoint_12_52_To$Snap") (<no position>)
thread 'rustc' panicked at 'explicit panic', viper/src/verifier.rs:203:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Is it clear for you how to proceed?

Reply:
I'm actually not quite sure. It could be that the return type of a function does not get encoded. The `$Snap` in the error message should be replaced with the correct type. Or the `$SortWrappers` is out of place, and I forgot to unwrap something.

Reply:
I'm actually not quite sure. It could be that the return type of a function does not get encoded. The `$Snap` in the error message should be replaced with the correct type. Or the `$SortWrappers` is out of place, and I forgot to unwrap something.